File: app/src/main/java/com/josycom/mayorjay/flowoverstack/viewmodel/AnswerViewModel.java
package com.josycom.mayorjay.flowoverstack.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.josycom.mayorjay.flowoverstack.model.Answer;
import com.josycom.mayorjay.flowoverstack.repository.AnswerRepository;

import java.util.List;

public class AnswerViewModel extends ViewModel {

    private LiveData<List<Answer>> mAnswersLiveData;

    AnswerViewModel(int questionId, String order, String sortCondition, String site, String filter) {
        AnswerRepository answerRepository = new AnswerRepository(questionId, order, sortCondition, site, filter);
        mAnswersLiveData = answerRepository.getAnswers();
    }

    public LiveData<List<Answer>> getAnswersLiveData() {
        return mAnswersLiveData;
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/viewmodel/AnswerViewModel.kt
package com.josycom.mayorjay.flowoverstack.viewmodel

import com.josycom.mayorjay.flowoverstack.data.repository.AnswerRepository
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import androidx.paging.PagingData
import androidx.paging.cachedIn
import com.josycom.mayorjay.flowoverstack.data.model.Answer
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

@HiltViewModel
class AnswerViewModel @Inject constructor(private val answerRepository: AnswerRepository) : ViewModel() {

    var answerDataFlow: Flow<PagingData<Answer>>? = null

    fun init(questionId: Int, order: String, sortCondition: String, site: String, filter: String, siteKey: String) {
        answerRepository.init(questionId, order, sortCondition, site, filter, siteKey)
        answerDataFlow = answerRepository.answerDataFlow?.cachedIn(viewModelScope)
    }
}